src/waveform/renderers/waveformmarkset.cpp

```diff
@@ -60,15 +65,27 @@ void WaveformMarkSet::setup(const QString& group, const QDomNode& node,
 
             if (controlItemSet.insert(hotCueControlItem).second) {
                 //qDebug() << "WaveformRenderMark::setup - Automatic mark" << hotCueControlItem;
-                m_marks.push_back(m_defaultMark);
-                WaveformMark& mark = m_marks.back();
-                mark.setKeyAndIndex(pHotcue->getKey(), i);
+                m_marks.push_back(new WaveformMark(m_pDefaultMark.get()));
+                WaveformMark* mark = m_marks.back();
+                mark->setKeyAndIndex(pHotcue->getKey(), i);
             }
         }
     }
 }
 
 void WaveformMarkSet::clear() {
-    m_defaultMark = WaveformMark();
+    m_pDefaultMark = std::make_unique<WaveformMark>();
+    qDeleteAll(m_marks);
     m_marks.clear();
 }
+
+const WaveformMark* WaveformMarkSet::getHotCueMark(int hotCue) {
+    DEBUG_ASSERT(hotCue >= 0);
+    DEBUG_ASSERT(hotCue < NUM_HOT_CUES);
+    return m_marks[m_iFirstHotCue + hotCue];
+}
+
+void WaveformMarkSet::setHotCueMark(int hotCue, WaveformMark* mark) {
+    delete m_marks[m_iFirstHotCue + hotCue];
```

Comment:
This will also invalidate the pointer formerly returned by getHotCueMark(hotCue)! Please use either shared pointers for this purpose or (much simpler) use value semantics if WaveformMark is a small object that is cheap to copy.